### PR TITLE
Early Perf benchmark for SchemaTransformer

### PR DESCRIPTION
**Pull request by tinnou**
Early results on my local machine (Macbook pro 2.4 GHz 8-Core Intel Core i9 32 GB 2667 MHz DDR4)

```
# Warmup: 1 iterations, 10 s each
# Measurement: 1 iterations, 10 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: benchmark.SchemaTransformerBenchmark.benchMarkSchemaTransformerRemove

# Run progress: 50.00% complete, ETA 00:02:59
# Fork: 1 of 1
# Warmup Iteration   1: 74264.966 ms/op
Iteration   1: 28894.776 ms/op


Result "benchmark.SchemaTransformerBenchmark.benchMarkSchemaTransformerRemove":
  28894.776 ms/op


# Run complete. Total time: 00:06:02

REMEMBER: The numbers below are just data. To gain reusable insights, you need to follow up on
why the numbers are the way they are. Use profilers (see -prof, -lprof), design factorial
experiments, perform baseline and negative tests that provide experimental control, make sure
the benchmarking environment is safe on JVM/OS/HW level, ask for reviews from the domain experts.
Do not assume the numbers tell you what you want them to tell.

Benchmark                                                    Mode  Cnt      Score   Error  Units
SchemaTransformerBenchmark.benchMarkSchemaTransformerAdd     avgt       30603.825          ms/op
SchemaTransformerBenchmark.benchMarkSchemaTransformerRemove  avgt       28894.776          ms/op
```

### CODE REVIEW

File: src/test/java/benchmark/SchemaTransformerBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            URL url = getResource(file);
+            return Resources.toString(url, Charsets.UTF_8);
+        }
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Measurement(iterations = 1, time = 10)
+    @Threads(1)
+    @Fork(1)
```
Comment (andimarek) [inline src/test/java/benchmark/SchemaTransformerBenchmark.java:119]: lets do 3 forks at least

File: src/test/java/benchmark/SchemaTransformerBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.google.common.io.Resources.getResource;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@Warmup(iterations = 2)
+@Measurement(iterations = 2, timeUnit = TimeUnit.NANOSECONDS)
```
Comment (andimarek) [inline src/test/java/benchmark/SchemaTransformerBenchmark.java:39]: I think these values are overwritten by the values below?

File: src/test/java/benchmark/SchemaTransformerBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        GraphQLSchema schema = myState.schema;
+        return SchemaTransformer.transformSchema(schema, myState.directiveAdder);
+    }
+
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Measurement(iterations = 1, time = 10)
+    @Threads(1)
+    @Fork(1)
```
Comment (andimarek) [inline src/test/java/benchmark/SchemaTransformerBenchmark.java:132]: same here: lets do 3 forks min

File: src/test/java/benchmark/SchemaTransformerBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        }
+
+        private String readFromClasspath(String file) throws IOException {
+            URL url = getResource(file);
+            return Resources.toString(url, Charsets.UTF_8);
+        }
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
```
Comment (andimarek) [inline src/test/java/benchmark/SchemaTransformerBenchmark.java:116]: lets do 2 warmup iterations

**Review by andimarek — CHANGES_REQUESTED** [review]
Thanks a lot!
See my comments.

File: src/test/java/benchmark/SchemaTransformerBenchmark.java (hunk 3 above)
Comment (tinnou) [reply src/test/java/benchmark/SchemaTransformerBenchmark.java:132]: roger that

**Pull request merged**